### DISCUSSION TITLE
Error creating a mesh

### DISCUSSION THREAD

**Discussion by Universe3d** [alicevision/Meshroom#1557] [Q&A]
I add circles of photos and on a certain circle from 3 to 5. The mesh begins for a very long time generated and the cube is obtained at the output, this error is constant on different scans, rebooting the program and the removal of the node does not help, an error has always appeared at different stages, but always on the last circles Photo. At the same time, the cloud points are perfect and shown correctly!
https://ibb.co/c2PBjjM

**Comment by natowi** [comment]
Do you have the model on a turntable or do you move the camera/rotate the transparent plate?

Did you try to import all complete circles in one go into Meshroom instead of augmenting?

How does your StructureFromMotion 3 look like? Is your meshed model within the "mesh cube"?

In general, you should get closer to your model, the image frame should be filled by your model. At the moment, your image contains 2/3 environment and 1/3 model:
![Unbenannt](https://user-images.githubusercontent.com/8138537/138588896-f3d68093-f209-4d22-b346-7770fad60c6a.png)

Also note, that flat objects are challenging to reconstruct in general.

**Comment by Universe3d** [comment]
1) This is a turning table yes
2) I add folders with circles as in the transference instruction, in order to be added a new node.
If you throw everything at once, the program is confused and the cloud of points is turned out to be bad, many frames are incorrectly positioned.
3) The model takes the highest possible amount when turning if I approach closer - then the model go beyond if it is horizontally - the depth of field will be worse on the edges.
4) This is not a specific example, just showed what kind of cube it turns out, I had models that occupied 60-80% of the frame and still this error climbed on a certain circle. An error appears on any model under any conditions
5) The node is the perfect cloud of points, and in the previous node mesh is generated correctly, and in this node cloud points even better than in the previous one.

Today I tried to change the `deapthmap` parameters, when the `downscale` decreases to the maximum value `16`, it can be seen that the program create the mesh correctly, but with such low quality) . As soon as I enhance the quality of the `downscale` `8-2`, the program starts the calculation very long literally 30 minutes, builds the grid and it turns out the cube with that low poly.

I have already deleted the cache because there are many photos weigh, I can not show you StructureFromMotion 3, but believe it is not a little worse, but for me even better than StructureFromMotion 2 on which everything is successful before texturing, of course a little bit more "flying" points, But the object and location of the cameras and all circles correctly!

Here the person here is the same problem and it can be seen that he does not have such a bad cloud of points, obviously not to get a cube - [https://github.com/alicevision/meshroom/issues/1502](https://github.com/alicevision/meshroom/issues/1502)

Yes, this complex object is really complicated for photogrammetry, but the cloud of points is perfect, just on this object, I again stumbled upon this strange in my opinion the problem .. I deleted the process, the model turned out to be good with two circles, it is a pity that the third circle could not improve Quality and capture parts from the upper angle.

Of course, the number of "flying" dot is increasing with each round, but even if the program usually builds them and it turns out just a bad model.
And here it turns out exactly the cube, it does not even coincide with the "flying" "wrong" dots.

**Reply by natowi** [reply]
Thank you for the detailed reply. We´ll look into it. I reopened the issue and fixed the link in your post.

**Comment by Universe3d** [comment]
According to experimentated with another model. It came to the conclusion that this error is still because of a large number of points that are background, with their definite exceeding - the model is built within their limits as a cube, in another software I tried the same model (Agisoft Metashape), in this program there are also there This is a cloud of white dots, but the program has masks and editing mode.
Even if in the third-party program to ignore this cloud of points, still does not work the cube, but simply the model with interference with individual polygons.
Made for myself the conclusion that all the same turning table and my photo quality is not good enough)
Also, the program lacks editing the clouds of points well or masks.
But with all this, the construction of the cube is very strange, and I tried to configure different parameters of the nodes in the entire node 4, nothing but I have already written a reduction in the quality of the mesh to the minimum value does not help (

https://ibb.co/yB4hpD9
https://ibb.co/X8pv5yC
https://ibb.co/KDW4KtZ
https://ibb.co/zbJk69y
https://ibb.co/F5WnvkN

_sorry for bad english, i use google translator))_
